src/gui/symbology/qgsgraduatedsymbolrendererwidget.cpp

```diff
@@ -798,6 +819,30 @@ void QgsGraduatedSymbolRendererWidget::methodComboBox_currentIndexChanged( int )
   }
 }
 
+void QgsGraduatedSymbolRendererWidget::updateMethodParameters()
+{
+  while ( mParametersLayout->rowCount() )
+    mParametersLayout->removeRow( 0 );
+  qDeleteAll( mParameterWidgetWrappers );
+
+  const QString methodId = cboGraduatedMode->currentData().toString();
+  QgsClassificationMethod *method = QgsApplication::classificationMethodRegistry()->method( methodId );
+  Q_ASSERT( method );
+
+  // todo need more?
+  QgsProcessingContext context;
+
+  for ( const auto *def : method->parameterDefinitions() )
+  {
+    QgsAbstractProcessingParameterWidgetWrapper *ppww = QgsGui::processingGuiRegistry()->createParameterWidgetWrapper( def, QgsProcessingGui::Standard );
+    mParametersLayout->addRow( ppww->createWrappedLabel(), ppww->createWrappedWidget( context ) );
```

Comment:
ppww->createWrappedLabel() may return null, e.g. with checkboxes. Is that an issue here?

Reply:
yes, for the log method I used a boolean.